internal/config/features.go

```diff
@@ -31,53 +33,57 @@ type features struct {
 	MOFED    *feature `toml:"mofed,omitempty"`
 	NVSWITCH *feature `toml:"nvswitch,omitempty"`
 	GDRCopy  *feature `toml:"gdrcopy,omitempty"`
+	// DisableIMEXChannelCreation ensures that the implicit creation of
+	// requested IMEX channels is skipped when invoking the nvidia-container-cli.
+	DisableIMEXChannelCreation *feature `toml:"disable-imex-channel-creation,omitempty"`
 }
 
 type feature bool
```

Comment:
Does it make simplify things if this was extended to:
```
type feature struct {
    enabled bool
    envvarOverride string
}
```

Then you can define  `isEnabled()` generically as:
```
func (f *feature) isEnabled(ins ...getenver) bool {
	if f == nil {
		return false
	}
	if f.envvar != "" {
		for _, in := range ins {
			if in.Getenv(envvar) == "enabled" {
				return true
			}
		}
	}
	return f.enabled
}
```

Reply:
Nevermind -- I see that the feature type is embedded directly in the config, so this wouldn't work.

Reply:
Yes, the intent is to have something like the following:
```
[features]
  disable-imex-channel-creation = true
  another-opt-in-feature = true
  another-feature-explicitly-disabled = true
```

Reply:
We could look into having a type per feature and then associating the envvar with the type instead, but I don't really see the benefit of that now.

Reply:
Thanks for the review @klueska. I have reworked the feature handling to introduce a type alias for each feature.

This adds a little more boiler plate, but removes the switch statement that we had. The other option that I was considering was to implement as custom unmarshaller for the `features` struct and extract the envvar from struct tags using reflection. The in-memory representation could then be something that you propose. I don't think that this is worth blocking on now though.